src/mesido/esdl/asset_to_component_base.py

```diff
@@ -743,7 +743,7 @@ def _get_connected_q_nominal(self, asset: Asset) -> Union[float, Dict]:
             q_nominals = {}
             try:
                 for port in asset.in_ports:
-                    connected_port = asset.in_ports[0].connectedTo[0]
```

Comment:
I think the else-statement doesn't make sense here, you might also have an inport with an electricity connection. It should only enter the else statement in case that it has looped over all in_ports and it never had an gas or heat commodity specified.

Reply:
I agree. This was already identified in the KIP gas work. Code has been updated